pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -98,6 +99,118 @@ type PagerDutyConfig struct {
 	HTTPConfig   *HTTPConfig             `json:"httpConfig,omitempty"`
 }
 
+// SlackConfig configures notifications via Slack.
+type SlackConfig struct {
+	SendResolved *bool                 `json:"sendResolved,omitempty"`
+	APIURL       *v1.SecretKeySelector `json:"apiURL,omitempty"`
+	Channel      *string               `json:"channel,omitempty"`
+	Username     *string               `json:"username,omitempty"`
+	Color        *string               `json:"color,omitempty"`
+	Title        *string               `json:"title,omitempty"`
+	TitleLink    *string               `json:"titleLink,omitempty"`
+	Pretext      *string               `json:"pretext,omitempty"`
+	Text         *string               `json:"text,omitempty"`
+	Fields       []SlackField          `json:"fields,omitempty"`
+	ShortFields  *bool                 `json:"shortFields,omitempty"`
+	Footer       *string               `json:"footer,omitempty"`
+	Fallback     *string               `json:"fallback,omitempty"`
+	CallbackID   *string               `json:"callbackId,omitempty"`
+	IconEmoji    *string               `json:"iconEmoji,omitempty"`
+	IconURL      *string               `json:"iconURL,omitempty"`
+	ImageURL     *string               `json:"imageURL,omitempty"`
+	ThumbURL     *string               `json:"thumbURL,omitempty"`
+	LinkNames    *bool                 `json:"linkNames,omitempty"`
+	MrkdwnIn     []string              `json:"mrkdwnIn,omitempty"`
+	Actions      []SlackAction         `json:"actions,omitempty"`
+	HTTPConfig   *HTTPConfig           `json:"httpConfig,omitempty"`
+}
+
+// Validate ensures SlackConfig is valid
+func (sc *SlackConfig) Validate() error {
+	for _, action := range sc.Actions {
+		if err := action.Validate(); err != nil {
+			return err
+		}
+	}
+	for _, field := range sc.Fields {
+		if err := field.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// SlackAction configures a single Slack action that is sent with each notification.
+// See https://api.slack.com/docs/message-attachments#action_fields and https://api.slack.com/docs/message-buttons
+// for more information.
+type SlackAction struct {
+	Type         string                  `json:"type,omitempty"`
+	Text         string                  `json:"text,omitempty"`
+	URL          string                  `json:"url,omitempty"`
+	Style        string                  `json:"style,omitempty"`
+	Name         string                  `json:"name,omitempty"`
+	Value        string                  `json:"value,omitempty"`
+	ConfirmField *SlackConfirmationField `json:"confirm,omitempty"`
+}
+
+// Validate ensures SlackAction is valid
+func (sa *SlackAction) Validate() error {
+	if sa.Type == "" {
+		return errors.New("missing type in Slack action configuration")
```

Comment:
Thanks a lot for your review :)
Is there a way to test the operator locally so I could debug the current testing issue otherwise, I could use some help to unlock the e2e situation :D